sympy/polys/rationaltools.py

```diff
@@ -83,3 +83,52 @@ def _together(expr):
         return expr
 
     return _together(sympify(expr))
+
+
+def thiele_interpolation(u, v, var=symbols('x'), simplify=True):
+    """
+    Build a rational function from a finite set of inputs in vector u
+    and their function values in vector v (both vectors must be of equal
+    lengths).
+
+    The interpolation algorithm works well with approximate floating point
+    inputs; it may also work with exact values but it is known to yield some
+    division by zero when values are equally spaced.
+
+    Examples
+    ========
+
+    >>> from sympypolys.rationaltools import thiele_interpolation as thiele
+
+    >>> thiele([1, 2, 5, 6], [10, 12, 11, 13])
+    (9*x**2 + 29*x - 238)/(8*x - 28)
+
+    >>> from sympy import cos
+    >>> from sympy.abc import x
+    >>> a = [ 0.05*i for i in range(32) ]
+    >>> b = [ cos(x) for x in a ]
+    >>> f = thiele(a, b, simplify=False)
+    >>> f.subs(x, 0.5) * 3
+    3.14159265358979
+
+    >>> thiele([1, 2, 3, 4], [10, 12, 14, 16])
```

Comment:
Would it be better to raise an exception in this case?

Reply:
Yes, but how can I achieve that? Not sure about the cleanest wai...

Reply:
I tried to implement it. Did I do it the right way with `expr.has(nan, zoo)`?

Reply:
@asmeurer Could you review my code and tell me if it is OK?